py/fiberassign/run.py

```diff
@@ -0,0 +1,997 @@
+    multiprocessing.
+
+    Args:
+        args (namespace): The parsed arguments.
+
+    Returns:
+        None
+
+    """
+    tiles, columns = run_merge_init(args)
+    merge_results(args.targets, tiles, result_dir=args.dir,
+                  result_prefix=args.prefix, result_split_dir=args.split,
+                  out_dir=args.out, out_prefix=args.out_prefix,
+                  out_split_dir=args.out_split, columns=columns,
+                  copy_fba=(not args.skip_raw))
+    return
+
+
+def run_merge_mpi(args, comm):
+    """Run output merging.
+
+    This uses the previously parsed options to read input data and perform
+    merging of the input catalogs.  This is designed to be run with one MPI
+    process per node.  Each MPI process in the input communicator will then
+    use multiprocessing for further parallelism.
+
+    Args:
+        args (namespace): The parsed arguments.
+        comm (MPI.Comm):  The MPI communicator.
+
+    Returns:
+        None
+
+    """
+    log = Logger.get()
+    tiles, columns = run_merge_init(args, comm)
+    ptiles = np.array_split(tiles, comm.size)[comm.rank]
+    if len(ptiles) > 0:
+        log.info("proc {} doing {} tiles".format(comm.rank, len(ptiles)))
+        merge_results(args.targets, ptiles, result_dir=args.dir,
+                      result_prefix=args.prefix, out_dir=args.out,
+                      out_prefix=args.out_prefix, columns=columns,
+                      copy_fba=(not args.skip_raw))
+    return
+
+
+def run_qa(args):
+    """Run QA.
+
+    This uses the previously parsed options to read input data and run the
+    QA.
+
+    Args:
+        args (namespace): The parsed arguments.
+
+    Returns:
+        None
+
+    """
+    hw, tiles = run_qa_init(args)
+    qa_tiles(hw, tiles, result_dir=args.dir, result_prefix=args.prefix,
+             result_split_dir=args.split, qa_out=args.out)
+    return
+
+
+def run_plot(args):
+    """Run plotting of fiberassign outputs.
+
+    This uses the previously parsed options to read input data and make
+    per-tile plots of the fiber assignment.
+
+    Args:
+        args (namespace): The parsed arguments.
+
+    Returns:
+        None
+
+    """
+    hw, tiles, petals = run_plot_init(args)
+    plot_tiles(hw, tiles, result_dir=args.dir, result_prefix=args.prefix,
+               result_split_dir=args.split, plot_dir=args.out,
+               plot_prefix=args.out_prefix, plot_split_dir=args.out_split,
+               petals=petals, real_shapes=args.real_shapes)
+    return
+
+
+def run_plot_qa(args):
+    """Run QA plotting.
+
+    This uses the previously parsed options to read input data and make a
+    plot of the QA results.
+
+    Args:
+        args (namespace): The parsed arguments.
+
+    Returns:
+        None
+
+    """
+    qadata = None
+    with open(args.qafile, "r") as f:
+        qadata = json.load(f)
+    plot_qa(qadata, args.outroot, labels=args.labels)
+    return
```

Comment:
We're at almost a 1000 lines for this file and it is mixing the command line parsing and run functions for multiple different scripts separated by hundreds of lines of code for the pieces that go together.  I'd prefer to organize these like desispec et al, with desispec/scripts/ and one module file per script file.